docs/expected_loss.ipynb

```diff
@@ -0,0 +1,203 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "prompt-victorian",
+   "metadata": {},
+   "source": [
+    "# Federations vs. simple risk splitting\n",
+    "\n",
+    "After a recent [Twitter discussion](https://twitter.com/liber_liver/status/1453654819227512834) I wanted to explore how much better federated custodians protect against malicious actors than naively splitting one's capital over multiple centralized custodians. We assume that each federation member/custodian has a probability $P_{mal}$ of being malicios.\n",
+    "\n",
+    "We are now interested in the expected loss $E(\\text{loss})$ a user suffers due to that. Loss is defined as losing access to their funds. In the federated setting that can be either due to theft or due to the adversary stopping the federation (which is easier). There exist countermeasures against adversaries merely stopping the federation (e.g. for ransom) but we will pessimistically assume that both failure modes mean loss of funds for the user (even if only for a limited time). In the centralized custodial case on the other hand the money gets stolen or not.\n",
+    "\n",
+    "We explore both federations of optimal sizes $(i \\cdot 3) + 1$ for as well as federations of suboptimal sizes. We see that adding members to a federation of already optimal size hurts resilience in our model since there are more potentially malicious parties but the threshold for how many malicious parties the system can handle does not change."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "prescribed-assumption",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def max_faulty_liveness(n):\n",
+    "    \"\"\"Returns maximum the number of malicious federation members for liveness\"\"\"\n",
+    "    return (n - 1) // 3\n",
+    "\n",
+    "def max_faulty_fund_safety(n):\n",
+    "    \"\"\"Returns maximum the number of malicious federation members for fund safety\"\"\"\n",
+    "    return n - max_faulty_liveness(n) - 1\n",
+    "\n",
+    "def p_mal_fail(n, mal, pmal):\n",
+    "    \"\"\"\n",
+    "    Returns the probability of mal of n members being malicious given the\n",
+    "    single-member-malicious probability pmal.\n",
+    "    \"\"\"\n",
+    "    return binomial(n, mal) * pmal**mal * (1 - pmal)**(n - mal)"
```

Comment:
Is the binomial distribution correct for this simulation? If the idea is to sample the federation members for malicious behavior, perhaps the [https://en.wikipedia.org/wiki/Hypergeometric_distribution](Hypergeometric Distribution) warrants a look?

Reply:
Thought about this quite a bit, but imo under the assumption that A being malicious does not affect B, C and D's probability of being malicious I still think it's binomial. Hypergeometric would apply if the probability changed iiuc. 

Actually in the first place I just drew a big decision tree of members being either malicious or honest and classified leafs based on malicious count. Only later I remembered that one can just use `nCr` (`binomial(n, r)` in sage) to model the number of different ways to choose.

Maybe there is a good argument to be made for changes in probability depending on other members though?

Reply:
The "sampling without replacement" part of the hypergeometric distribution would apply if federation members were excluded from future rounds after having been deemed malicious in an earlier round. But that assumes the client can detect malicious members. Is that possible?

In the one-shot case, binomial seems correct. Maybe this analysis could be extended later to a multi-round simulation?

Reply:
I think the optimal strategy for an adversary is to wait till they have compromised enough members to achieve their goal. It does not make sense to reveal themselves earlier, they can just act like honest members till then.

Reply:
Agreed. Then the analysis looks correct as-is.